minifi_main/JsonSchema.cpp

```diff
@@ -0,0 +1,436 @@
+            | ranges::views::join(std::string{','})
+            | ranges::to<std::string>())
+        << "]";
+
+  out << R"(, "properties": {)";
+  for (size_t prop_idx = 0; prop_idx < props.size(); ++prop_idx) {
+    const auto& property = props[prop_idx];
+    if (prop_idx != 0) out << ",";
+    writePropertySchema(property, out);
+  }
+  out << "}";  // "properties"
+  out << "}";  // "Properties"
+}
+
+static std::string buildSchema(const std::unordered_map<std::string, std::string>& relationships, const std::string& processors, const std::string& controller_services) {
+  std::stringstream all_rels;
+  for (const auto& [name, rels] : relationships) {
+    all_rels << "\"relationships-" << escape(name) << "\": " << rels << ", ";
+  }
+
+  std::stringstream remote_port_props;
+  writeProperties(minifi::RemoteProcessorGroupPort::properties(), minifi::RemoteProcessorGroupPort::SupportsDynamicProperties, remote_port_props);
+
+  std::string process_group_properties = R"(
+    "Processors": {
+      "type": "array",
+      "items": {"$ref": "#/$defs/processor"}
+    },
+    "Connections": {
+      "type": "array",
+      "items": {"$ref": "#/$defs/connection"}
+    },
+    "Controller Services": {
+      "type": "array",
+      "items": {"$ref": "#/$defs/controller_service"}
+    },
+    "Remote Process Groups": {
+      "type": "array",
+      "items": {"$ref": "#/$defs/remote_process_group"}
+    },
+    "Process Groups": {
+      "type": "array",
+      "items": {"$ref": "#/$defs/simple_process_group"}
+    },
+    "Funnels": {
+      "type": "array",
+      "items": {"$ref": "#/$defs/funnel"}
+    }
+  )";
+
+  std::stringstream cron_pattern;
+  {
+    const char* all = "\\\\*";
+    const char* any = "\\\\?";
+    const char* increment = "(-?[0-9]+)";
+    const char* secs = "([0-5]?[0-9])";
+    const char* mins = "([0-5]?[0-9])";
+    const char* hours = "(1?[0-9]|2[0-3])";
+    const char* days = "([1-2]?[0-9]|3[0-1])";
+    const char* months = "([0-9]|1[0-2]|jan|feb|mar|apr|may|jun|jul|aug|sep|oct|nov|dec)";
+    const char* weekdays = "([0-7]|sun|mon|tue|wed|thu|fri|sat)";
+    const char* years = "([0-9]+)";
+
+    auto makeCommon = [&] (const char* pattern) {
+      std::stringstream common;
+      common << all << "|" << any
+        << "|" << pattern << "(," << pattern << ")*"
+        << "|" << pattern << "-" << pattern
+        << "|" << "(" << all << "|" << pattern << ")" << "/" << increment;
+      return std::move(common).str();
+    };
+
+    cron_pattern << "^"
+      << "(" << makeCommon(secs) << ")"
+      << "( " << makeCommon(mins) << ")"
+      << "( " << makeCommon(hours) << ")"
+      << "( " << makeCommon(days) << "|LW|L|L-" << days << "|" << days << "W" << ")"
+      << "( " << makeCommon(months) << ")"
+      << "( " << makeCommon(weekdays) << "|L|" << weekdays << "#" << "[1-5]" << ")"
+      << "( " << makeCommon(years) << ")?"
+      << "$";
```

Comment:
```suggestion
    cron_pattern << "(?i)^"
      << "(" << makeCommon(secs) << ")"
      << " (" << makeCommon(mins) << ")"
      << " (" << makeCommon(hours) << ")"
      << " (" << makeCommon(days) << "|LW|L|L-" << days << "|" << days << "W" << ")"
      << " (" << makeCommon(months) << ")"
      << " (" << makeCommon(weekdays) << "|" << weekdays << "?L|" << weekdays << "#" << "[1-5]" << ")"
      << "( (" << makeCommon(years) << "))?"
      << "$";
```
based on our debugging session I think this should be the regex